File: convert_icon.js
const fs = require('fs');
const path = require('path');
const { Resvg } = require('@resvg/resvg-js');

const sizes = {
    'mdpi': 48,
    'hdpi': 72,
    'xhdpi': 96,
    'xxhdpi': 144,
    'xxxhdpi': 192
};

const svgPath = 'Jamrah-Icon.svg';
const baseResDir = path.join('android', 'app', 'src', 'main', 'res');

async function convert() {
    try {
        const svgContent = fs.readFileSync(svgPath);

        for (const [density, size] of Object.entries(sizes)) {
            console.log(`Converting ${density} -> ${size}x${size}`);
            const mipmapDir = path.join(baseResDir, `mipmap-${density}`);
            if (!fs.existsSync(mipmapDir)) {
                fs.mkdirSync(mipmapDir, { recursive: true });
            }
            const outPath = path.join(mipmapDir, 'ic_jamrah.png');

            const resvg = new Resvg(svgContent, {
                fitTo: { mode: 'width', value: size },
            });
            const pngData = resvg.render().asPng();
            fs.writeFileSync(outPath, pngData);
        }
        console.log("Done generating launcher icons.");
    } catch (e) {
        console.error("Error generating icons:", e);
    }
}

convert();

File: convert_icon.py
import os
import cairosvg

# Sizes for launcher icons
sizes = {
    'mdpi': 48,
    'hdpi': 72,
    'xhdpi': 96,
    'xxhdpi': 144,
    'xxxhdpi': 192
}

svg_path = 'Jamrah-Icon.svg'
base_res_dir = r'android\app\src\main\res'

for density, size in sizes.items():
    mipmap_dir = os.path.join(base_res_dir, f'mipmap-{density}')
    os.makedirs(mipmap_dir, exist_ok=True)
    out_path = os.path.join(mipmap_dir, 'ic_jamrah.png')
    
    print(f'Converting {density} -> {size}x{size}')
    try:
        cairosvg.svg2png(url=svg_path, write_to=out_path, output_width=size, output_height=size)
    except Exception as e:
        print(f"Error: {e}")

print("Done generating launcher icons.")
